lib/shoulda/matchers/action_controller/route_params.rb

```diff
@@ -26,17 +28,24 @@ def controller_and_action_given_as_string?
         def extract_params_from_string
           controller, action = args[0].split('#')
           params = (args[1] || {}).merge(controller: controller, action: action)
-          stringify_values(params)
+          normalize_values(params)
         end
 
         def stringify_params
-          stringify_values(args[0])
+          normalize_values(args[0])
         end
 
-        def stringify_values(hash)
-          hash.inject({}) do |hash_copy, (key, value)|
-            hash_copy[key] = stringify(value)
-            hash_copy
+        def normalize_values(hash)
+          hash.each_with_object({}) do |(key, value), hash_copy|
+            hash_copy[key] = symbolize_or_stringify(key, value)
+          end
+        end
+
+        def symbolize_or_stringify(key, value)
+          if key.in?(VALUES_TO_SYMBOLIZE)
+            value.to_sym
```

Comment:
I could go either way. This is fine.

I wish there were a better way to say this actually. The only reason we stringify is for the params I believe, the other "special" params like controller, action, format, etc. don't need to be touched. Ah well.